modules/recon/contacts-credentials/hibp_paste.py

```diff
@@ -53,16 +53,19 @@ def module_run(self, accounts):
                         self.alert(f"Download not available for {paste['Source']} pastes.")
                     self.alert(f"{account} => Paste found! Seen in a {paste['Source']} on {paste['Date']} ({fileurl}).")
                     if download == True:
-                        resp = self.request(fileurl)
-                        if resp.status_code == 200:
-                            filepath = f"{self.workspace}/{_safe_file_name(fileurl)}.txt"
-                            if not os.path.exists(filepath):
-                                dl = open(filepath, 'w')
-                                dl.write(resp.text.encode(resp.encoding) if resp.encoding else resp.text)
-                                dl.close()
-                            self.verbose(f"Paste stored at '{filepath}'.")
-                        else:
-                            self.alert(f"Paste could not be downloaded ({fileurl}).")
+                        try:
+                            resp = self.request(fileurl)
+                            if resp.status_code == 200:
+                                filepath = f"{self.workspace}/{_safe_file_name(fileurl)}.txt"
+                                if not os.path.exists(filepath):
+                                    dl = open(filepath, 'w')
+                                    dl.write(resp.text.encode(resp.encoding).decode('utf-8') if resp.encoding else resp.text)
+                                    dl.close()
+                                self.verbose(f"Paste stored at '{filepath}'.")
+                            else:
+                                self.alert(f"Paste could not be downloaded ({fileurl}).")
+                        except Exception as e:
```

Comment:
What was causing the issue here? Was there a specific exception class? If it had to do with the request itself, please only wrap that line of code with the `try` block and be as specific as possible with the exception. Thanks.

Reply:
Also, it looks like the code from the pushpin PR made it into this one too. Please clean up the commit when you make the above changes. Thanks!